src/functional/ffront/type_info.py

```diff
@@ -286,6 +312,37 @@ def is_concretizable(symbol_type: ct.SymbolType, to_type: ct.SymbolType) -> bool
     return False
 
 
+def _is_empty_field(field: ct.SymbolType) -> bool:
+    return isinstance(field, ct.FieldType) and len(field.dims) == 0
+
+
+def is_not_empty_field_compatible(a_arg: ct.SymbolType, b_arg: ct.SymbolType) -> bool:
+    """
+    Check if first argument is an empty field and whether second is either another emtpy field or a scalar.
```

Comment:
Have to think about this, but the description is very technical and doesn't enhance what is understood by simply reading the code.

Reply:
I made it slightly less technical